src/Microsoft.DotNet.Build.Tasks/GetPerfTestAssemblies.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public sealed class GetPerfTestAssemblies : Task
+    {
+        [Required]
+        public ITaskItem[] TestBinaries { get; set; }
+
+        [Output]
+        public ITaskItem[] PerfTestAssemblies { get; set; }
+
+        public override bool Execute()
+        {
+            Log.LogMessage(MessageImportance.High, "About to inspect {0} test assemblies.", TestBinaries.Length);
+            var perfTests = new List<ITaskItem>();
+
+            foreach (var testBinary in TestBinaries)
+            {
+                Log.LogMessage(MessageImportance.Low, "Inspecting assembly {0}.", testBinary.ItemSpec);
+                Assembly assembly = Assembly.ReflectionOnlyLoadFrom(testBinary.ItemSpec);
```

Comment:
We should not be using ReflectionOnlyLoad. We should use the MetadataReader for this activity.